src/main/java/com/google/api/codegen/transformer/SampleImportTransformer.java

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.metacode.InitCodeNode;
+import com.google.api.codegen.viewmodel.CallingForm;
+import com.google.api.codegen.viewmodel.ImportSectionView;
+import com.google.api.codegen.viewmodel.OutputView;
+import java.util.List;
+
+public class SampleImportTransformer {
```

Comment:
It makes sense to make it an interface. I am not sure, however, if it should extend `ImportSectionTransformer`, since the concrete classes of this interface won't need to implement methods defined in `ImportSectionTransformer`. For example, they wouldn't be able to generate an `ImportSection` from a collection of `InitCodeNodes`.

Reply:
It would work, but I have a few concerns:
- There are two methods in `ImportSectionTransformer`:
```java
ImportSectionView generateImportSection(TransformationContext context, String className);
ImportSectionView generateImportSection(MethodContext context, Iterable<InitCodeNode> specItemNodes);
```
And if we have `SampleImportTransformer` extend `ImportSectionTransformer`, we will need to disallow users from calling the first method, probably by throwing an `UnsupportedOperationException` in `StandardSampleImportTransformer`, since that method wouldn't make sense for the import section for samples. So is it still worth extending `ImportSectionTransformer`?

- I am fine with the idea of having callers calling whatever `addXXX` to add imports in certain parts of the samples (that actually provides a lot of flexibility for in code samples in the future potentially). But if we make this decision on the surface, it seems better to make `generateImportSection` like:
```java
ImportSectionView generateImportSection(MethodContext context);
// or
ImportSectionView generateImportSection(ImportTypeTable typeTable);
```
Because all we need if all the types added to that `typeTable`. Reusing the method `mportSectionView generateImportSection(MethodContext context, Iterable<InitCodeNode> specItemNodes)` seems to indicated that the `SampleImportTransformer` generates an `ImportSection` from the `methodContext` and the `specItemNodes`, while the latter should have been taken care of by `addInitCodeImports`.

I am also not sure if these concerns are actually serious enough, but happy to know your thoughts.

Reply:
I think those concerns are serious enough. Let's try making it an interface but not extending `ImportSectionTransformer`, as you suggested previously.